pkg/services/grafana-apiserver/common.go

```diff
@@ -33,6 +34,11 @@ type APIGroupBuilder interface {
 
 	// Get the API routes for each version
 	GetAPIRoutes() *APIRoutes
+
+	// Optionally add an authorization hook
+	// Standard namespace checking will happen before this is called, specifically
+	// the namespace must matches an org|stack that the user belongs to
+	GetAuthorizer() authorizer.AuthorizerFunc
 }
 
```

Comment:
thoughts on splitting this out into a different interface if it's optional?
```suggestion
}

// Optionally add an authorization hook
// Standard namespace checking will happen before this is called, specifically
// the namespace must matches an org|stack that the user belongs to
type APIGroupAuthorizerProvider interface {
	GetAuthorizer() authorizer.Authorizer
}
```